### PR TITLE
add new benchmark test for sdatmcor

### PR DESCRIPTION
**Pull request by nschweig-nrao**
Added benchmark tests suggested by Takeshi on CAS-13982

### CODE REVIEW

File: benchmarks/bench_sdatmcor.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    def tearDown(self):
+        smart_remove(self.infile)
+        smart_remove(self.outfile)
+        smart_remove(self.caltable)
+
+    def skip_omp_test_if_darwin(self):
+        sysname = os.uname()[0]
+        if sysname == 'Darwin':
+            self.skipTest('Skip OpenMP tests on macOS')
+            return
```
Comment (amcnicho) [inline benchmarks/bench_sdatmcor.py:66]: Good to know if some subset of tests are not expected to pass on macOS, but doesn't seem necessary here

File: benchmarks/bench_sdatmcor.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    def time_sdatmcor_explicit_atmtype(self):
+        """Test specifying atmtype explicitly."""
+        sdatmcor(infile=self.infile, outfile=self.outfile, datacolumn='data', atmtype=2)
+    time_sdatmcor_explicit_atmtype.version = "CAS-13982"
+
+    def time_sdatmcor_gainfactor_dict(self):
+        """Test gainfactor: dict input."""
+        gainfactor = {'19': 10.0, '23': 45.0}
+        apply_gainfactor(self.infile, 19, gainfactor['19'])
+        apply_gainfactor(self.infile, 23, gainfactor['23'])
```
Comment (amcnicho) [inline benchmarks/bench_sdatmcor.py:82]: Should these calls be in the setup method?

**Review by amcnicho — COMMENTED** [review]
I think this new test script should have an explicit iteration control block at the top, as we've done with other files. Appreciate the explicit versioning. ~5m to run in my sandbox. Apart from minor tweaks it looks fine to merge.

**Comment by nschweig-nrao** [comment]
I've made the changes suggested to the branch and it's still up-to-date with master. I'll go ahead with the merge now.

**Pull request merged**